### ISSUE TITLE
Version 2

### ISSUE THREAD

**Issue by cgcardona** [Bitcoin-com/rest.bitcoin.com#48]
The time has come to start work on REST v2. This is an issue to capture high level deliverables and action items. We can link to item-specific PRs and Issues below.

Work has started on this in the [`/v2` directory](https://github.com/Bitcoin-com/rest.bitcoin.com/tree/stage/routes/v2)

## Refactor `GET` to `POST`

* https://github.com/Bitcoin-com/rest.bitcoin.com/issues/49

## Remove URL params from `POST`

* https://github.com/Bitcoin-com/rest.bitcoin.com/issues/50
* Once this is complete them merge [Smart Sam's `wormhole-sdk`](https://github.com/Bitcoin-com/wormhole-sdk/pull/65/files) changes.

## Test refactor
  - unit
  - integration

## Reject testnet addresses on mainnet and vice-versa

* https://github.com/Bitcoin-com/rest.bitcoin.com/issues/47

## Improved Error Handling and Logging

* [Error Handling](https://github.com/Bitcoin-com/rest.bitcoin.com/issues/29)
* [Logging](https://github.com/Bitcoin-com/rest.bitcoin.com/issues/59)

## Misc

* authentication
* dynamic rate limit
* token balances with utxos
* websockets / events
* If tx is 'already in block chain' return txid instead
* add `/wh` to the url for WH related endpoints
* logging

**Comment by christroutner** [comment]
I'd like us to move the tests from mainnet to testnet. 

The Wormhole REST tests were originally designed to target testnet, but REST v1 targeted mainnet. We decided to switch the Wormhole REST tests over to mainnet when we merged it into REST.

Using mainnet has a lot of disadvantages, primarily in that it costs real money. Our tests can be much more elaborate; spanning unit, integration, and end-to-end tests if we focus on testnet.

**Comment by cgcardona** [comment]
I don't see how using testnet saves us money. We shouldn't be spending money each time we run our test suite.

Even if we use testnet we shouldn't be creating real WH tokens each time we run our test suite.

**Comment by christroutner** [comment]
You're right. My logic is incorrect above.

The main difference is when unexpected behavior occurs. If we need to manually verify and interrogate mainnet tests, it costs money. But if we focus on testnet, this isn't an issue.

**Comment by christroutner** [comment]
PR #73 kicks off this refactor. See the PR description for more details.

**Comment by christroutner** [comment]
I created Issue #74 to discuss the implementation of pagination.

**Comment by christroutner** [comment]
Created Issue #98 to fix WH error handling.

**Comment by christroutner** [comment]
Clarity needed for Issue #104 before I can continue with the Block and Blockchain libraries.

**Comment by cgcardona** [comment]
V2 is now live. We can create a new issue if there is any cleanup work we need.
